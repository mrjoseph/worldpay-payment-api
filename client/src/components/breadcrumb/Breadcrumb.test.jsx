import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import BreadcrumbWithRouter from './Breadcrumb';

describe('Breadcrumb', () => {
  const props = {
    breadcrumb: {
      checkout: 'checkout',
      payment: 'payment',
      authorized: 'Authorized',
    },
  };

  it('Should render the breadcrumb with props', () => {
    const rendered = renderer.create(
      <Router>
        <BreadcrumbWithRouter {...props} />
      </Router>,
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
