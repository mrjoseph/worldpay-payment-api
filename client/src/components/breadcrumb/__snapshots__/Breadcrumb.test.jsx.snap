// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb Should render the breadcrumb with props 1`] = `
<nav
  aria-label="breadcrumb"
>
  <ol
    className="breadcrumb"
  >
    <li
      className="breadcrumb-item"
    >
      <a
        href="/checkout"
        onClick={[Function]}
      >
        <button
          className="button"
          type="button"
        >
          checkout
        </button>
      </a>
    </li>
    <li
      className="breadcrumb-item"
    >
      <a
        href="/payment"
        onClick={[Function]}
      >
        <button
          className="button"
          type="button"
        >
          payment
        </button>
      </a>
    </li>
    <li
      className="breadcrumb-item"
    >
      <a
        href="/authorized"
        onClick={[Function]}
      >
        <button
          className="button"
          type="button"
        >
          Authorized
        </button>
      </a>
    </li>
  </ol>
</nav>
`;
